pymc/distributions/continuous.py

```diff
@@ -650,9 +653,9 @@ def dist(
         cls,
         mu: Optional[DIST_PARAMETER_TYPES] = None,
         sigma: Optional[DIST_PARAMETER_TYPES] = None,
-        tau: Optional[DIST_PARAMETER_TYPES] = None,
         lower: Optional[DIST_PARAMETER_TYPES] = None,
         upper: Optional[DIST_PARAMETER_TYPES] = None,
+        tau: Optional[DIST_PARAMETER_TYPES] = None,
```

Comment:
We shouldn't change the order of arguments. We can however make everything after sigma kwarg only, since the position of `tau` is indeed strange.

`def dist(cls, mu, sigma, *, tau, lower ...)`

Reply:
Better to do this in a separate PR so that it can be reviewed on it's own

Reply:
Ok, I'll take care of creating the new PR and cleaning this up